code/modules/vore/resizing/resize_vr.dm

```diff
@@ -180,82 +180,94 @@ var/const/RESIZE_A_SMALLTINY = (RESIZE_SMALL + RESIZE_TINY) / 2
 		if(D.flying) //if the prey is flying, don't smush them.
 			return
 
-	if(src.a_intent == I_DISARM && src.canmove && !src.buckled)
-		// If bigger than them by at least 0.75, move onto them and print message.
-		if((src.get_effective_size() - tmob.get_effective_size()) >= 0.75)
+	if(!src.canmove ||src.buckled)
+		return
+
+	if(src.a_intent == I_HELP)
+		if(handle_micro_bump_helping(tmob))
+			forceMove(tmob.loc)
 			now_pushing = 0
+			return 1
+
+	else if(src.a_intent == I_DISARM)
+		// Need to be at least twice as large
+		if((src.get_effective_size() / tmob.get_effective_size()) >= 2)
 			src.forceMove(tmob.loc)
-			if(src.m_intent == "run") //Running down the hallway with disarm intent?
-				tmob.resting = 1 //Force them down to the ground.
+			now_pushing = 0
+			tmob.resting = 1 // Force them down to the ground.
+			tmob.Weaken(1) // Updates lying, canmove and icons
+			if(src.m_intent == "run")
 				var/mob/living/carbon/human/H = src
 				if(istype(H) && istype(H.tail_style, /datum/sprite_accessory/tail/taur/naga))
 					src << "Your tail slides over [tmob], pushing them down to the ground!"
-					tmob << "[src]'s tail slides over you, forcing you down to the ground!"
+					tmob << "<span class='warning'>[src]'s tail slides over you, forcing you down to the ground!</span>"
 				else
-					src << "You quickly push [tmob] to the ground with your foot!"
-					tmob << "[src] pushes you down to the ground with their foot!"
+					src << "You push [tmob] to the ground with your foot!"
+					tmob << "<span class='warning'>[src] pushes you down to the ground with their foot!</span>"
+				log_and_message_admins("has pushed [tmob] to the ground by stepping on them.") //Both humans and mobs, since stepping on mobs can be abused.
 				admin_attack_log(src, tmob, "Pinned [tmob.name] under foot.", "Was pinned under foot by [src.name].", "Pinned [tmob.name] under foot.")
 				return 1
-			if(src.m_intent == "walk") //Most likely intentionally stepping on them.
-				var/size_damage_multiplier = (src.size_multiplier - tmob.size_multiplier)
-				var/damage = (rand(15,30)* size_damage_multiplier) //Since stunned is broken, let's do this. Rand 15-30 multiplied by 1 min or 1.75 max. 15 holo to 52.5 holo, depending on RNG and size differnece.
+			else // Most likely intentionally stepping on them.
+				var/size_damage_multiplier = 0.25 * (src.size_multiplier / tmob.size_multiplier)
```

Comment:
Before:
var/size_damage_multiplier = (src.size_multiplier - tmob.size_multiplier)
Max of 1.75, min of 0.5

New:
var/size_damage_multiplier = 0.25 * (src.size_multiplier / tmob.size_multiplier)
2.0/1.5 = 1.333 * 0.25 = .325
1.5/1 = 1.5 * 0.25 = 0.375
1.0/0.5 = 2.0 * 0.25 = 0.5
0.75/0.25 = 3.0 * 0.25 = .75
Min of .325 and max of .75. This means damage from halloss is greatly nerfed. Not only that, but 2.0 can't step on 2.0 can't step on 1.5, and 1.5 also can't step on 1.0 as shown above.

Reply:
Your calculations are a bit odd here.
```
Maximum: 2   = 0.25 * max_size/min_size
Minimum: 0.5 = 0.25 * 2

The fact that a 2.0 can't step on a 1.25 is lent to the fact that the size now requires to be twice the value. Back when using discrete values however, the size of 1.25 did not exist, so applying the same logic to it does not make sense here.

Reply:
>now requires to be twice the value

I really don't like this. This isn't a tweak, but a complete change.

The calculations I did to calculate the damage are fine as long as I can see. I changed it to be (src.size_multiplier / tmob.size_multiplier) * 0.25 to make it easier to read.

Reply:
Redid the math, I see the mistakes I made here.

.25 * (2.0/1.5) = .33333333333 (Unable to do so with this nerf.)
.25 * (2.0/1.0) = .5
.25 * (2.0/.75) = .6666666666
.25 * (2.0/.5) = 1
.25 * (2.0/.25) = 2
.25 * (1.5/1.0) = .375 (Unable to do with this nerf)
.25 * (1.5/.75) = .5
.25 * (1.5/.5) = .75
.25 * (1.5/.25) = 1.5
.25 * (1.0/.5) = .5
.25 * (1.0/.25) = 1
.25 * (.75/.25) = .75
.25 * (.5/.25) = .5

So the lower damage is still the same, and the upper damage is slightly buffed. This I'm fine with, but _the fact 200% can't pick up 150% or _anyone above 100%, including 101% for that matter_ is fucking sneaky and not ok whatsoever.